pages/platform/data-processing/36_HOWTO_use_object_storage_java/guide.en-asia.md

```diff
@@ -0,0 +1,190 @@
+---
+title: How to use object storage with Apache Spark on the Data Processing platform using Java
+slug: object-storage-java
+excerpt: Object storage based on Swift and its S3 API is the common way to store data. Let’s find out how to do it in Java!
+section: How to
+order: 6
+---
+
+**Last updated 19<sup>th</sup> May, 2020**
+
+## Objective
+This guide gives you a basic example about using OpenStack Swift and its S3 API with OVHcloud Data Processing using Java.
+
+We will use the OpenStack S3 API to read and write data to Swift Object Storage.
+
+Samples are based on the well-known WordCount. We will first read data from a text file, then count the frequence of each word in this particular file. And the print the result in output log and also write the result in a text file in Swift object storage. 
+
+## Requirements
+- Access to the [OVHcloud Control Panel](https://www.ovh.com/auth/?action=gotomanager){.external}.
+- An OVHcloud account.
+- A cloud project in your OVHcloud account (see [How to create a cloud project](../../public-cloud/getting_started_with_public_cloud_logging_in_and_creating_a_project){.external} for details).
+- Data Processing activated (see [How to activate the Data Processing](../activation){.external} for details).
+- The input file 'novel.txt' used by the following samples is available [Here](https://github.com/ovh/data-processing-samples/tree/master/java_wordCount_S3/novel.txt){.external}.
+You have to upload it in a Swift container. (see [Getting started with the Swift API](../../public-cloud/getting_started_with_the_swift_api/){.external} to read more about how to create a Swift container).
+- OVHcloud object storage "secret key" and "access key" for the container that you have uploaded the 'novel.txt' text file. (See [How to create EC2 credentials](../../public-cloud/getting_started_with_the_swift_S3_API/){.external} for more details).
+
+## Read/Write data with Apache Spark using OpenStack Swift S3 API in Java
```

Comment:
Why is that the only tutorial ? Shouldn't it match 1 to 1 the different chapters from the 35_HOWTO doc ?

Reply:
because every object in the container that contains the code, will be copied to the cluster and then they can read it like a local file and there is no need to any configuration or credentials for swift or s3 api. But they can not then write into the swift like a local file. Because the files written in cluster will not copied back to the original container. So I think it is better to use always the S3 API to read and write from containers to avoid and mistake and confusion.